File: src/main/java/org/havenapp/main/service/AlertSender.java
package org.havenapp.main.service;

import java.util.ArrayList;

public interface AlertSender {

    public void setUsername (String username);

    public void reset ();

    public void register (boolean callEnabled);

    public void verify (final String verificationCode);

    public void stopHeartbeatTimer ();

    public void startHeartbeatTimer (int countMs);

    public void sendMessage (final ArrayList<String> recipients, final String message, final String attachment);

}

File: src/main/java/org/havenapp/main/senders/AlertSender.kt
package org.havenapp.main.senders

import java.util.*

interface AlertSender {
    fun setUsername(username: String?)
    fun reset()
    fun register()
    fun verify(verificationCode: String?)
    fun stopHeartbeatTimer()
    fun startHeartbeatTimer(countMs: Int)
    fun sendMessage(
        recipients: ArrayList<String?>?, message: String?,
        attachment: String?
    )
}